test/end-to-end/signup.test.js

```diff
@@ -476,6 +476,75 @@ describe("SuperTokens SignUp", function () {
                 throw new Error("test failed");
             }
         });
+
+        it("Check on blank form submit nonOptionalErrorMsg gets displayed as expected", async function () {
+            await submitForm(page);
+            let formFieldErrors = await getFieldErrors(page);
+            // Also standard non-optional-error is displayed if nonOptionalErrorMsg is not provided
+            assert.deepStrictEqual(formFieldErrors, [
+                "Field is not optional",
+                "Field is not optional",
+                "You must accept the terms and conditions",
+            ]);
```

Comment:
you should also enforce that there is no network call being made, and that this error is due to frontend checking.